packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -8,80 +8,31 @@
-#include <Teuchos_CommandLineProcessor.hpp>
-#include <Teuchos_StandardCatchMacros.hpp>
-
-#include <Kokkos_DefaultNode.hpp>
-
-#include <DTK_ConfigDefs.hpp>
 
 #include <DTK_LinearBVH.hpp>
 
-#include <cmath>
-#include <functional>
-#include <random>
-
-namespace details = DataTransferKit::Details;
-
-std::vector<std::array<double, 3>>
-make_stuctured_cloud( double Lx, double Ly, double Lz, int nx, int ny, int nz )
-{
-    std::vector<std::array<double, 3>> cloud( nx * ny * nz );
-    std::function<int( int, int, int )> ind = [nx, ny]( int i, int j, int k ) {
-        return i + j * nx + k * ( nx * ny );
-    };
-    double x, y, z;
-    for ( int i = 0; i < nx; ++i )
-        for ( int j = 0; j < ny; ++j )
-            for ( int k = 0; k < nz; ++k )
-            {
-                x = i * Lx / ( nx - 1 );
-                y = j * Ly / ( ny - 1 );
-                z = k * Lz / ( nz - 1 );
-                cloud[ind( i, j, k )] = {{x, y, z}};
-            }
-    return cloud;
-}
+#include <Kokkos_DefaultNode.hpp>
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
 
-std::vector<std::array<double, 3>> make_random_cloud( double Lx, double Ly,
-                                                      double Lz, int n )
-{
-    std::vector<std::array<double, 3>> cloud( n );
-    std::default_random_engine generator;
-    std::uniform_real_distribution<double> distribution_x( 0.0, Lx );
-    std::uniform_real_distribution<double> distribution_y( 0.0, Ly );
-    std::uniform_real_distribution<double> distribution_z( 0.0, Lz );
-    for ( int i = 0; i < n; ++i )
-    {
-        double x = distribution_x( generator );
-        double y = distribution_y( generator );
-        double z = distribution_z( generator );
-        cloud[i] = {{x, y, z}};
-    }
-    return cloud;
-}
+#include <chrono>
+#include <cmath> // cbrt
+#include <random>
 
 template <class NO>
 int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
 {
     using DeviceType = typename NO::device_type;
     using ExecutionSpace = typename DeviceType::execution_space;
 
-    double Lx = 100.0;
-    double Ly = 100.0;
-    double Lz = 100.0;
-    int nx = 11;
-    int ny = 11;
-    int nz = 11;
-    int n_points = 100;
-    std::string mode = "radius";
+    int n_values = 50000;
+    int n_queries = 20000;
+    int n_neighbors = 10;
 
-    clp.setOption( "nx", &nx, "source mesh points in x-direction." );
-    clp.setOption( "ny", &ny, "source mesh points in y-direction." );
-    clp.setOption( "nz", &nz, "source mesh points in z-direction." );
-    clp.setOption( "N", &n_points,
-                   "number of target mesh points (distributed randomly)." );
-    clp.setOption( "mode", &mode, "mode: (knn | radius)" );
+    clp.setOption( "values", &n_values, "number of indexable values (source)" );
+    clp.setOption( "queries", &n_queries, "number of queries (target)" );
```

Comment:
I think the language is a bit confusing. I tend to think about the problem in source point and target points. But you think in terms of values and queries. That's foreign to me, but probably fine.